internal/command/deploy/deploy_build.go

```diff
@@ -234,17 +239,73 @@ func determineImage(ctx context.Context, appConfig *appconfig.Config, useWG, rec
 	return
 }
 
+// downloadDockerfileFromURL downloads a Dockerfile from a URL to a temporary file
+func downloadDockerfileFromURL(ctx context.Context, dockerfileURL string) (tempPath string, err error) {
+	// Create a temporary file for the Dockerfile
+	tempFile, err := os.CreateTemp("", "dockerfile-*.tmp")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temporary file: %w", err)
+	}
+	defer tempFile.Close()
+
+	// Download the Dockerfile
+	resp, err := http.Get(dockerfileURL)
+	if err != nil {
+		os.Remove(tempFile.Name())
```

Comment:
Can you move os.Remove() to defer? You can check err in defer to see whether the function is successfully finished or not.